src/main/java/eiteam/esteemedinnovation/commons/init/ContentModule.java

```diff
@@ -93,66 +119,67 @@ protected Item setup(Item startingItem, String path) {
-    }
-
-    /**
-     * Overload for {@link #setup(Block, String, CreativeTabs, Function)} that uses {@link EsteemedInnovation#tab}
-     * as the tab.
-     */
-    protected Block setup(Block startingBlock, String path, Function<Block, ItemBlock> itemBlockFunc) {
-        return setup(startingBlock, path, EsteemedInnovation.tab, itemBlockFunc);
+    protected Block setup(RegistryEvent.Register<Block> event, Block startingBlock, String path) {
+        return setup(event, startingBlock, path, EsteemedInnovation.tab);
     }
 
     /**
      * Sets up a Block with an unlocalized name, a registry name, an optional creative tab, and registers it to the
      * Block registry.
+     * @param event The event to register on
      * @param startingBlock The block to start with
      * @param path The name of the block, excluding the EI mod ID
      * @param tab The creative tab to add it to. Use null to not add it to any tab. You will have to cast the null to
      *            {@link CreativeTabs}.
-     * @param itemBlockFunc A function that passes a {@link Block} and returns an {@link ItemBlock}. The returned value
-     *                      is what will be used as the block's according item. Pass null to make this Block not have
-     *                      any item. You will have to cast the null to {@link Function}.
      * @return The registered block (ItemBlock is not returned).
      */
-    protected Block setup(Block startingBlock, String path, CreativeTabs tab, Function<Block, ItemBlock> itemBlockFunc) {
+    protected Block setup(RegistryEvent.Register<Block> event, Block startingBlock, String path, CreativeTabs tab) {
         startingBlock.setUnlocalizedName(Constants.EI_MODID + ":" + path);
         if (tab != null) {
             startingBlock.setCreativeTab(tab);
         }
         startingBlock.setRegistryName(Constants.EI_MODID, path);
-        GameRegistry.register(startingBlock);
-        if (itemBlockFunc != null) {
-            ItemBlock ib = itemBlockFunc.apply(startingBlock);
-            ib.setRegistryName(startingBlock.getRegistryName());
-            GameRegistry.register(ib);
-        }
+        event.getRegistry().register(startingBlock);
         return startingBlock;
     }
 
+    protected void setupItemBlock(RegistryEvent.Register<Item> event, Block startingBlock) {
+        setupItemBlock(event, startingBlock, EsteemedInnovation.tab);
+    }
+
+    protected void setupItemBlock(RegistryEvent.Register<Item> event, Block startingBlock, Function<Block, ItemBlock> itemBlock) {
+        setupItemBlock(event, startingBlock, EsteemedInnovation.tab, itemBlock);
+    }
+
+    protected void setupItemBlock(RegistryEvent.Register<Item> event, Block startingBlock, CreativeTabs tab) {
+        setupItemBlock(event, startingBlock, tab, ItemBlock::new);
+    }
+
+
+    protected void setupItemBlock(RegistryEvent.Register<Item> event, Block startingBlock, CreativeTabs tab, Function<Block, ItemBlock> itemBlock) {
```

Comment:
That is there just to make registering `ItemBlock`s easier. They are also different from Items in the sense that they don't need the registry name (It'll take from the `Block`) and they also need the block they represent. So these are here mainly for helping register the `ItemBlock`s.

Reply:
Hm. It might be an idea to have a `Map` of `Block => Function<Block, ItemBlock>` that is added to in `registerBlock` so we can still automatically register `ItemBlock`s without having to manually do it. What do you think?

Reply:
Hmm. I guess, but the way I can see the implementation of that would only move code that you have in the `registerItems` method into the creation of that map. It wouldn't really change the amount of work required.